gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -504,6 +506,208 @@ def _get_variant_calls(self, record, formats):
+               representative_header_lines=None,  # type:  List[str]
+               splittable_bgzf=False,  # type: bool
+               **kwargs  # type: **str
+              ):
+    # type: (...) -> None
+    super(PySamParser, self).__init__(file_name,
+                                      range_tracker,
+                                      file_pattern,
+                                      compression_type,
+                                      allow_malformed_records,
+                                      representative_header_lines,
+                                      **kwargs)
+    # These members will be properly initiated in _init_with_header().
+    self._vcf_reader = None
+    self._to_child = None
+    self._original_info_list = None
+
+  def _init_parent_process(self, p_read, c_write, c_read, p_write):
+    # Parent Process
+    os.close(c_read)
+    os.close(c_write)
+    into_pysam = os.fdopen(p_read)
+    self._to_child = os.fdopen(p_write, 'w')
+    self._vcf_reader = libcbcf.VariantFile(into_pysam, 'r')
+    self._original_info_list = self._vcf_reader.header.info.keys()
+
+  def _write_header_lines(self, to_pysam, header_lines):
+    # Feed the data with a break line, since we read data line by line - if
+    # we would use the conventional ``read()`` method, both pipes would lock,
+    # while waiting for an EOF signal.
+    for header in header_lines:
+      to_pysam.write(header + '\n')
+    to_pysam.flush()
+
+  def _write_to_second_pipe(self, text_line):
+    self._to_child.write(text_line.encode('utf-8') + '\n')
+    self._to_child.flush()
+
+  def _read_from_second_pipe(self, from_parent):
+    return from_parent.readline()
+
+  def _write_to_first_pipe(self, to_pysam, text_line):
+    to_pysam.write(text_line)
+    to_pysam.flush()
+
+  def _read_from_first_pipe(self):
+    return next(self._vcf_reader)
+
+  def _init_child_process(self, p_read, c_write, c_read, p_write, header_lines):
+    # Child process' task is to populate data into the pipe that feeds
+    # VariantFile class - first by populating all of the header lines, and then
+    # by redirecting the data received from the second pipe.
+
+    os.close(p_read)
+    os.close(p_write)
+    from_parent = os.fdopen(c_read)
+    to_pysam = os.fdopen(c_write, 'w')
+
+    self._write_header_lines(to_pysam, header_lines)
+
+    while True:
+      text_line = self._read_from_second_pipe(from_parent)
+      if not text_line:
+        break
+      self._write_to_first_pipe(to_pysam, text_line)
+
+    from_parent.close()
+    to_pysam.close()
+    sys.exit(0)
+
+  def _init_with_header(self, header_lines):
+    # PySam requires a version header to be present, so add one if absent.
+    if header_lines and not header_lines[0].startswith(
+        FILE_FORMAT_HEADER_TEMPLATE.format(VERSION='')):
+      header_lines.insert(0, FILE_FORMAT_HEADER_TEMPLATE.format(VERSION='4.0'))
+
+    # First pipe is responsible for supplying data from child process into the
```

Comment:
pysam_read, pysam_write ?

variants_read, variants_write ?

I'd explain here that the reason for the variants pipe is that we don't have access to the pysam handle to call into the library from the get_variant function because it's a different process.  It's not really about lines or anything else, it's just that we can't call into pysam from that process.